src/engraving/dom/navigate.cpp

```diff
@@ -190,8 +191,29 @@ ChordRest* nextChordRest(const ChordRest* cr, bool skipGrace, bool skipMeasureRe
 
     track_idx_t track = cr->track();
     SegmentType st = SegmentType::ChordRest;
+    Segment* curSeg = cr->segment();
+    Score* score = cr->score();
+    for (Segment* seg = curSeg->next1MM(st); seg; seg = seg->next1MM(st)) {
+        if (disableOverRepeats) {
+            // Disallow inputting ties between unrelated voltas
+            // This visually adjacent segment is never the next to be played
```

Comment:
I can barely picture in my head what this scenario looks like, but that's fine, you know this inside out so I trust you on that. The important thing is that we _have_ to find a way to move this bit of code into a function of its own (peraps in `utils.cpp` if we can't find any other suitable place). This code is repeated like 5 or 6 times throughout the PR. If you ever need to change it, your future self will hate you with passion 😅